clients/drcachesim/tools/view.cpp

```diff
@@ -292,13 +306,17 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
         printed_header_.find(memref.marker.tid) == printed_header_.end()) {
         printed_header_.insert(memref.marker.tid);
         if (trace_version_ != -1) { // Old versions may not have a version marker.
-            if (!should_skip(memstream, memref)) {
+            if (!should_skip(memstream, memref) &&
+                // Do not print if user skipped headers.
+                filetype_from_record_) {
```

Comment:
filetype is the first record expected in the headers and that's why we're checking that? Add comment, and maybe rename s/filetype_from_record_/skipped_headers/?